File: src/main/java/ru/romanow/serialization/model/XmlTestObject.java
package ru.romanow.serialization.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import javax.xml.bind.annotation.*;
import java.io.Serializable;
import java.util.List;

@Data
@Accessors(chain = true)
@EqualsAndHashCode(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class XmlTestObject
        extends TestObject
        implements Serializable {
    private static final long serialVersionUID = 632112641149051390L;

    @XmlElement(name = "message")
    protected String message;

    @XmlElement(name = "code")
    protected Integer code;

    @XmlElement(name = "status")
    protected Status status;

    @XmlElement(name = "innerData")
    protected InnerData innerData;

    @XmlElementWrapper(name = "publicDataList")
    protected List<PublicData> publicData;
}

File: src/main/java/ru/romanow/serialization/model/XmlTestObject.kt
package ru.romanow.serialization.model

import jakarta.xml.bind.annotation.*
import org.apache.commons.lang3.RandomStringUtils.randomAlphanumeric
import org.apache.commons.lang3.RandomUtils
import java.io.Serializable

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
open class XmlTestObject(
    @XmlElement(name = "message")
    override var message: String? = null,

    @XmlElement(name = "code")
    override var code: Int? = null,

    @XmlElement(name = "status")
    override var status: Status? = null,

    @XmlElement(name = "innerData")
    override var innerData: InnerData? = null,

    @XmlElementWrapper(name = "publicDataList")
    override var publicData: List<PublicData>? = null
) : TestObject(), Serializable {

    override fun toString(): String {
        return "XmlTestObject(message=$message, code=$code, status=$status, innerData=$innerData, publicData=$publicData)"
    }
}

fun createXmlTestObject() =
    XmlTestObject(
        message = randomAlphanumeric(10),
        code = RandomUtils.nextInt(0, 100),
        status = Status.DONE,
        innerData = buildInnerData(),
        publicData = listOf(buildPublicData(), buildPublicData())
    )